crates/typst-library/src/model/heading.rs

```diff
@@ -267,18 +267,29 @@ impl Show for Packed<HeadingElem> {
             realized = realized.styled(ParElem::set_hanging_indent(indent.into()));
         }
 
-        Ok(if html {
+        if html {
             // HTML's h1 is closer to a title element. There should only be one.
             // Meanwhile, a level 1 Typst heading is a section heading. For this
             // reason, levels are offset by one: A Typst level 1 heading becomes
             // a `<h2>`.
-            let level = self.resolve_level(styles);
-            let t = [tag::h2, tag::h3, tag::h4, tag::h5, tag::h6][level.get().min(5) - 1];
-            HtmlElem::new(t).with_body(Some(realized)).pack().spanned(span)
+            let level = self.resolve_level(styles).get();
+            if level >= 6 {
+                engine.sink.warn(warning!(span, "heading of level {} should be transformed to <h{}> in HTML, but only <h1> to <h6> are supported; instead transformed to <div role=\"heading\" aria-level=\"{}\">, but note that this is not supported by all screen readers", level, level + 1, level + 1;
+                    hint: "you may want to restructure your document so that it doesn't contain deep headings"));
+                Ok(HtmlElem::new(tag::div)
+                    .with_body(Some(realized))
+                    .with_attr(const { HtmlAttr::constant("role") }, "heading")
```

Comment:
this works, but I think it's nicer to add these to the macro in `mod attr` in `dom.rs` (which creates a constant just like this). then we can have them all in one place, meaning less chance for typos (this won't be the only `role` attr we need). 

as for aria level: the macro doesn't support case transform, but maybe we can just spell out the constants with a hyphen manually next to the macro, at least for now. or we can leave it here, since it's unlikely to be reused anyway.